client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -108,6 +136,72 @@ const EventForm: React.FC<EventFormProps> = (props) => {
           </FormControl>
         )}
 
+        <FormControl id="first-name" isRequired>
+          <Box display="flex" alignItems="end" m="1">
+            <FormLabel> Sponsers</FormLabel>
+            <Spacer />
+            <Button
+              onClick={() => {
+                append({
+                  type: 'FOOD',
+                  id: 0,
+                });
+              }}
+            >
+              Add
+            </Button>
+          </Box>
+          {sponsors.map((sponsor, index) => {
+            return (
+              <Flex key={sponsor.key} borderWidth="1px" p="5" mb="5">
+                <Box display="flex" flexGrow={1}>
+                  <FormControl m="1">
+                    <FormLabel>Sponsor Type</FormLabel>
+                    <Select
+                      defaultValue={getValues(`sponsors.${index}.type`)}
+                      {...register(`sponsors.${index}.type` as const, {
+                        required: true,
+                      })}
```

Comment:
```suggestion
                     {...registeredSponsor}
                      onChange={(e) => {
                        registeredSponsor.onChange(e);
                        const sponsorsForThisType =
                          sponsorData?.sponsors.filter(
                            (s) => s.type === e.target.value,
                          ) ?? [];
                        setValue(
                          `sponsors.${index}.id`,
                          sponsorsForThisType[0]?.id,
                        );
                      }}
```
I noticed an issue that if you *just* changed the type and didn't click on the name, the form would not register the name correctly.